### PR TITLE
Study class format edited and worked on #8

### PR DESCRIPTION
**Pull request by joheeju**
1. Study class
- Matched all variables with lower_case_with_underscores
- Changed some format of arguments (For example, contact, age, etc) according to raw data

2. #8  
- Worked on almost everything except some vague parts (will discuss during meeting on 9/18)

### CODE REVIEW

**Comment by adeeb897** [comment]
What is BeautifulSoup? If that's from a package, we should start a requirements.txt with this pull request to keep track of packages and make them easier to install.

**Comment by adeeb897** [comment]
Also, are there other packages I need to install, because I'm not able to run this code

**Comment by jkim3389** [comment]
> What is BeautifulSoup? If that's from a package, we should start a requirements.txt with this pull request to keep track of packages and make them easier to install.

its crawling tool so that you can scrap the website easily

**Comment by adeeb897** [comment]
> > What is BeautifulSoup? If that's from a package, we should start a requirements.txt with this pull request to keep track of packages and make them easier to install.
> 
> its crawling tool so that you can scrap the website easily

So we have to install it using pip? What's the package name

**Comment by jkim3389** [comment]
> > > What is BeautifulSoup? If that's from a package, we should start a requirements.txt with this pull request to keep track of packages and make them easier to install.
> > 
> > 
> > its crawling tool so that you can scrap the website easily
> 
> So we have to install it using pip? What's the package name

Yeap so what I installed is
pip3 install BeautifulSoup4
pip3 install requests
pip3 install lxml

and in code 
from bs4 import BeautifulSoup to use it

**Comment by adeeb897** [comment]
> > > > What is BeautifulSoup? If that's from a package, we should start a requirements.txt with this pull request to keep track of packages and make them easier to install.
> > > 
> > > 
> > > its crawling tool so that you can scrap the website easily
> > 
> > 
> > So we have to install it using pip? What's the package name
> 
> Yeap so what I installed is
> pip3 install BeautifulSoup4
> pip3 install requests
> pip3 install lxml
> 
> and in code
> from bs4 import BeautifulSoup to use it

Are you able to run the code after that? I'm still getting compilation errors.

Also we should put something like this

beautifulsoup4==4.9.1
lxml==4.5.2
requests==2.24.0
urllib3==1.25.10

into a requirements.txt then we can just do `pip install -r requirements.txt`

File: crawler.py [hunk 1]
```diff
@@ -1,39 +1,56 @@
...
         self.title = title
         self.description = description
         self.last_updated = last_updated
         self.ID = ID
-        self.type = sType
+        self.type = study_type
         self.conditions = conditions
         self.sponsor = sponsor
-        self.recruitmentStatus = recruitmentStatus
+        self.recruitment_status = recruitmentStatus
```
Comment (adeeb897) [inline crawler.py]: `recruitmentStatus` needs to be `recruitment_status`

File: crawler.py [hunk 2]
```diff
@@ -1,39 +1,56 @@
...
         self.age = age
         self.sex = sex
         self.control = control
-        self.additionalCriteria = additionalCriteria
+        self.additional_criteria = additionalCriteria
```
Comment (adeeb897) [inline crawler.py]: `additionalCriteria` needs to be `additional_criteria`

File: crawler.py [hunk 3]
```diff
@@ -1,39 +1,56 @@
 from typing import List, Dict
 from datetime import datetime as date, time
-
+import urllib.request
```
Comment (adeeb897) [inline crawler.py:3]: I think we need to add import?
`from bs4 import BeautifulSoup`

**Review by adeeb897 — CHANGES_REQUESTED** [review]
I wasn't able to compile your code, but I think I noticed all of the compilation errors. Were you able to compile and run your code?

File: crawler.py [hunk 4]
```diff
@@ -41,11 +42,11 @@ def __init__(self, title: str, description: str, last_updated: date, ID: str, st
         self.type = study_type
         self.conditions = conditions
         self.sponsor = sponsor
-        self.recruitment_status = recruitmentStatus
+        self.recruitment_status = recruitment_Status
```
Comment (adeeb897) [inline crawler.py]: `recruitment_status`

**Review by adeeb897 — CHANGES_REQUESTED** [review]

File: crawler.py [hunk 5]
```diff
@@ -93,46 +78,44 @@ def download_and_format(id: str) -> Study:
...
         
+    sponsor = data.find("lead_sponsor").find("agency").get_text() ##lead sponsor only (no collborators)
+    recruitment_status = data.find("overall_status").get_text()
+            
+    age = {"min":data.find("minimum_age").get_text(), "max":data.find("maximum_age").get_text()}
+            
     sex = data.find("gender").get_text()
-    control = "" #TODO #??
-    additional_criteria = "" #TODO #inclusion/exclusion
+    control = data.find("healthy_volunteers").get_text()
```
Comment (adeeb897) [inline crawler.py]: This part caused an error when I tested your code on  random studies. I recommend doing this


```suggestion
try:
       control = data.find("healthy_volunteers").get_text()
except AttributeError:
       control = None
```

File: crawler.py [hunk 6]
```diff
@@ -93,46 +78,44 @@ def download_and_format(id: str) -> Study:
...
-    c = data.find("overall_contact")
-    if c is not None:
-        contact["name"] = c.find("last_name").get_text()
-        contact["phone"] = c.find("phone").get_text()
-        contact["email"] = c.find("email").get_text()
+    if data.findAll("location") is not None:
+        for loc in data.findAll("location"):
+            localLocation = loc.find("name").get_text()
+            nationalLocation = loc.find("address").get_text().strip("\n").replace("\n", ", ")
+            locations.append({"locationLocation": localLocation, "nationalLocation": nationalLocation, "status": status})
```
Comment (adeeb897) [inline crawler.py]: The status part also caused an error, do we need it? maybe we can do something like this
```suggestion
            locations.append({"locationLocation": localLocation, "nationalLocation": nationalLocation})
```

**Review by adeeb897 — CHANGES_REQUESTED** [review]
The changes look good, it's compiling for me! There are some parts causing run-time errors though. but you might have not seen these errors since I tested them on random studies. We should be able to fully test this code next week for similar errors.

I commented on which lines caused problems, once those are fixed we should be good!

By the way you can test your code by doing something like
`
s = download_and_format("NCT04554966")`
`print(vars(s))
`
at the very bottom and then running the python script

**Review by adeeb897 — COMMENTED** [review]
Looks good!

**Review by adeeb897 — APPROVED** [review]

**Pull request merged**